pkg/codegen/dotnet/gen.go

```diff
@@ -684,6 +684,7 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) error {
 	if r.DeprecationMessage != "" {
 		fmt.Fprintf(w, "    [Obsolete(@\"%s\")]\n", strings.Replace(r.DeprecationMessage, `"`, `""`, -1))
 	}
+	fmt.Fprintf(w, "    [ResourceIdentifier(\"%s\", \"%s\")]\n", r.Token, mod.pkg.Version.String())
```

Comment:
Unfortunately, I don't think we can emit the version like this as we would have already generated the code and committed it before the version has been tagged in Git.

For Node and Python, we've looked it up at runtime, just like how we do inside the resource constructors.

https://github.com/pulumi/pulumi/blob/a5c30f1547c9865d55f6aaab4bb4c434d29b4202/pkg/codegen/nodejs/gen.go#L1408

https://github.com/pulumi/pulumi/blob/a5c30f1547c9865d55f6aaab4bb4c434d29b4202/pkg/codegen/python/gen.go#L534

Should we do the same here, using the same `Utilities.Version` used inside the resource constructors?

https://github.com/pulumi/pulumi/blob/a5c30f1547c9865d55f6aaab4bb4c434d29b4202/pkg/codegen/dotnet/gen.go#L820

If we still want the version to come from the attribute, would we unseal it and make the `Version` property `virtual`, and then generate a subclass attribute in the package that overrides the `Version` to return `Utilities.Version`, and then apply the subclass attribute throughout the package's resources?

I guess an alternative would be to use a string replacement, and at build time first go through all .cs files for resources and replace a placeholder value with the actual version, but that'd involve changing a bunch of files.

Reply:
Right, thanks for that. I got confused by this line https://github.com/pulumi/pulumi/blob/223f6501eb826f6020e7209b205e3e61359eccd6/pkg/codegen/dotnet/gen.go#L1442 (I guess I should get rid of it)

Reply:
I think a runtime lookup is fine. Are we okay requiring every resource-containing assembly to require a class called `Utilities` with a property `Version`? That's true for all our providers, for now.